packages/sources/view-function/src/transport/function.ts

```diff
@@ -0,0 +1,70 @@
+import { Transport, TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { AdapterRequest, AdapterResponse } from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/function'
+import { ethers, utils } from 'ethers'
+
+export type FunctionTransportTypes = BaseEndpointTypes
+
+export class FunctionTransport implements Transport<FunctionTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<FunctionTransportTypes>
+  requester!: Requester
+  provider!: ethers.providers.JsonRpcProvider
+
+  async initialize(
+    dependencies: TransportDependencies<FunctionTransportTypes>,
+    _adapterSettings: FunctionTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.requester = dependencies.requester
+    this.name = transportName
+    this.provider = new ethers.providers.JsonRpcProvider(
+      _adapterSettings.ETHEREUM_RPC_URL,
+      _adapterSettings.ETHEREUM_CHAIN_ID,
+    )
+  }
+
+  async foregroundExecute(
```

Comment:
I think we may want to move this to a background execute. There could be a chance that a certain contract call takes a while to respond which could cause timeouts downstream even if the adapter doesn't fail to respond.